src/fstab-generator/fstab-generator.c

```diff
@@ -434,8 +434,7 @@ static int add_mount(
 
         /* Order the mount unit we generate relative to the post unit, so that DefaultDependencies= on the
          * target unit won't affect us. */
-        if (post && !FLAGS_SET(flags, MOUNT_AUTOMOUNT) && !FLAGS_SET(flags, MOUNT_NOAUTO) &&
-            !FLAGS_SET(flags, MOUNT_NOFAIL))
+        if (post && !FLAGS_SET(flags, MOUNT_NOFAIL))
                 fprintf(f, "Before=%s\n", post);
```

Comment:
i am not sure i follow.

So in the automount case the goal is that the actual mount triggered by the automount point unit can take any time it wants, and we don't delay the boot for it. otoh the automount point itself should be properly established during boot.

Thus this change doesn't look right to me, because it means that the mounts established by automounts no are synced against local-fs.target too. 


Dropping the check for "auto" might be OK otoh

Reply:
This is not for the automount unit, but the mount unit that will/might normally be started/triggered after boot. (Although IIRC the automount units themselves will have this ordering as well, but since they won't wait for the devices anyway, so it won't have this undesirable side-effect like in the nofail case.) This is entirely for the ordering during shutdown.

I'm not sure if there are some conrner case in which the automount mount unit will be triggerred during `/boot` (like, using automount on something like `/var`? But then that would seem to be misconfiguration to me.).

Reply:
For automount mounts: with this change they become ordered `Before=local-fs.target`. This means that if the mount is triggered *before* we reach `local-fs.target`, we'll delay `local-fs.target` for it. This is something that we don't want. It also means that the mount will only be stopped after `local-fs.target` is stopped, i.e. late in shutdown. This is the part you want, and which I think is quite reasonable. We want to delay unmounting, because otherwise we might unmount it, and then it might be triggered again by something during shutdown. It seems better to avoid that.

But I think that the ordering with `local-fs.target` might not matter. It only matters if *something* triggered the mount, and that something usually will be one of the early-boot services. If they are running before `local-fs.target`, they are probably ordered before it anyway, so the added ordering doesn't make much difference in practice.

And the logic is similar for `noauto`. In principle something could start a job for a noauto mount and now it'll be ordered earlier, but I don't think this would matter most of the time.

I think that the shutdown issues are real and handling shutdown better is more important than (somewhat theoretical) boot slowdowns.

Hmm, other opionions?

Reply:
Or in other words: if the effect of the added ordering is only in removing some parallelism during boot, then I think we should merge it, as discussed above. For me the important question is whether we could break something by introducing this additional ordering. I can imagine theoretical constructions like this, but I wonder if anyone sees real examples.

Reply:
> This means that if the mount is triggered before we reach local-fs.target, we'll delay local-fs.target for it.
But here's the thing: Is that true when the mount is neither wanted nor required by `local-fs.target`? (which should be the case with `noauto` or `x-systemd.automount`?)

Reply:
> This means that if the mount is triggered before we reach local-fs.target, we'll delay local-fs.target for it. This is something that we don't want.

For the record, one can still specify `nofail` additionally to invalidate that. (And at the same time, `nofail` remains to be the only case that is "on the loose" when it comes to shutdown.)